Automattic/wp-calypso
# Update/premium content loading assets code cleanup

This PR build on https://github.com/Automattic/wp-calypso/pull/44825

#### Changes proposed in this Pull Request

- moved asset `file exists` test to init function
- defined constants regarding assets dependencies and version so that they are easily reusable
- removed get_script_asset as a result of the above
- used PREMIUM_CONTENT__PLUGIN_DIR instead of $dir
- removed $index_js declaration
- moved paywall init to the bottom just before the filter

#### Testing instructions



Nothing should change after applying these modifications in terms of functionality. Confirm the Premium Content block is still working rightly.

Instructions on how to properly test the premium-content block here: p58i-8VC-p2
